servicetalk-client-api/src/main/java/io/servicetalk/client/api/ConnectionFactoryFilter.java

```diff
@@ -76,6 +77,8 @@
     /**
      * {@inheritDoc}
      *
+     * <p>If the returned strategy extends {@link ConnectExecutionStrategy} then the connection creation or accept may
+     * be offloaded.
      * <p>If the returned strategy extends {@code HttpExecutionStrategy} then the HTTP execution strategy will be
      * applied to the connections created.</p>
      * <p>A utility class provides the ability to combine connect and HTTP execution strategies,
```

Comment:
can you explain scenarios where users are expected to combine the different strategies? I would expect that is only something we do internally while computing the execution strategies.

If this can be limited to an internal implementation detail (e.g. via `merge`) then we can make `ConnectAndHttpExecutionStrategy` internal (and remove the reference in this comment).

Reply:
In the simplest case we would only need the `ConnectExecutionStrategy` to specify whether the connection factory, filters or acceptor requires offloading.

The additional capability `ConnectAndHttpExecutionStrategy` is needed for cases where the connection filter modifies the handling of request or otherwise changes the behaviour of the connection. It is equivalent to the connection filter installing a streaming filter per connection.